avogadro/io/lammpsformat.cpp

```diff
@@ -0,0 +1,413 @@
+    getline(inStream, buffer);
+    // Read z_min, z_max, tiltfactor_yz
+    vector<string> box_bounds_z(split(buffer, ' '));
+    z_min = lexicalCast<double>(box_bounds_z.at(0));
+    z_max = lexicalCast<double>(box_bounds_z.at(1));
+    tilt_yz = lexicalCast<double>(box_bounds_z.at(2));
+
+    x_min -= std::min(std::min(std::min(tilt_xy, tilt_xz), tilt_xy + tilt_xz),
+                      (double)0);
+    x_max -= std::max(std::max(std::max(tilt_xy, tilt_xz), tilt_xy + tilt_xz),
+                      (double)0);
+    y_min -= std::min(tilt_yz, (double)0);
+    y_max -= std::max(tilt_yz, (double)0);
+  }
+
+  // Else if unit cell is orthogonal, tilt factors are zero
+  else if (buffer.find("ITEM: BOX BOUNDS") == 0) {
+    // Read x_min, x_max
+    getline(inStream, buffer);
+    vector<string> box_bounds_x(split(buffer, ' '));
+    x_min = lexicalCast<double>(box_bounds_x.at(0));
+    x_max = lexicalCast<double>(box_bounds_x.at(1));
+    // Read y_min, y_max
+    getline(inStream, buffer);
+    vector<string> box_bounds_y(split(buffer, ' '));
+    y_min = lexicalCast<double>(box_bounds_y.at(0));
+    y_max = lexicalCast<double>(box_bounds_y.at(1));
+    // Read z_min, z_max
+    getline(inStream, buffer);
+    vector<string> box_bounds_z(split(buffer, ' '));
+    z_min = lexicalCast<double>(box_bounds_z.at(0));
+    z_max = lexicalCast<double>(box_bounds_z.at(1));
+  }
+
+  typedef map<string, unsigned char> AtomTypeMap;
+  AtomTypeMap atomTypes;
+  unsigned char customElementCounter = CustomElementMin;
+
+  // x,y,z stand for the coordinate axes
+  // s stands for scaled coordinates
+  // u stands for unwrapped coordinates
+  // scale_x = 0. if coordinates are cartesian and 1 if fractional (scaled)
+  getline(inStream, buffer);
+  vector<string> labels(split(buffer, ' '));
+  for (size_t i = 0; i < labels.size(); i++) {
+    if (labels[i] == "x" || labels[i] == "xu") {
+      x_idx = i;
+      scale_x = 0.;
+    } else if (labels[i] == "xs" || labels[i] == "xsu") {
+      x_idx = i;
+      scale_x = 1.;
+    } else if (labels[i] == "y" || labels[i] == "yu") {
+      y_idx = i;
+      scale_y = 0.;
+    } else if (labels[i] == "ys" || labels[i] == "ysu") {
+      y_idx = i;
+      scale_y = 1.;
+    } else if (labels[i] == "z" || labels[i] == "zu") {
+      z_idx = i;
+      scale_z = 0.;
+    } else if (labels[i] == "zs" || labels[i] == "zsu") {
+      z_idx = i;
+      scale_z = 1.;
+    } else if (labels[i] == "type")
+      type_idx = i;
+    else if (labels[i] == "id")
+      id_idx = i;
+  }
+
+  // Parse atoms
+  for (size_t i = 0; i < numAtoms; ++i) {
+    getline(inStream, buffer);
+    vector<string> tokens(split(buffer, ' '));
+
+    if (tokens.size() < labels.size() - 2) {
+      appendError("Not enough tokens in this line: " + buffer);
+      return false;
+    }
+
+    unsigned char atomicNum(0);
+    atomicNum =
+      static_cast<unsigned char>(lexicalCast<short int>(tokens[type_idx - 2]));
```

Comment:
I don't understand the lexicalCast *and* the static_cast - did the lexicalCast<unsigned char> fail?